alvr/server_core/src/connection.rs

```diff
@@ -244,16 +246,86 @@ pub fn handshake_loop(ctx: Arc<ConnectionContext>, lifecycle_state: Arc<RwLock<L
         }
     };
 
+    let mut adb_server_started = false;
+
     while *lifecycle_state.read() != LifecycleState::ShuttingDown {
-        dbg_connection!("handshake_loop: Try connect to manual IPs");
+        dbg_connection!("handshake_loop: Try connect to wired device");
 
-        let available_manual_client_ips = {
-            let mut manual_client_ips = HashMap::new();
-            for (hostname, connection_info) in SESSION_MANAGER
+        let mut wired_client_ips = HashMap::new();
+        if let Some((client_hostname, _)) =
+            SESSION_MANAGER
                 .read()
                 .client_list()
                 .iter()
-                .filter(|(_, info)| info.connection_state == ConnectionState::Disconnected)
+                .find(|(hostname, info)| {
+                    info.connection_state == ConnectionState::Disconnected
+                        && hostname.as_str() == WIRED_CLIENT_HOSTNAME
+                })
+        {
+            let layout = match FILESYSTEM_LAYOUT.get() {
+                None => {
+                    error!("Failed to get filesystem layout");
+                    thread::sleep(RETRY_CONNECT_MIN_INTERVAL);
+                    continue;
+                }
+                Some(layout) => layout,
+            };
+            let stream_port = SESSION_MANAGER.read().settings().connection.stream_port;
+            let status = match alvr_adb::setup_wired_connection(
+                layout,
+                CONTROL_PORT,
+                stream_port,
+                |downloaded, maybe_total| {
+                    if let Some(total) = maybe_total {
+                        alvr_events::send_event(EventType::Adb(alvr_events::AdbEvent {
+                            download_progress: downloaded as f32 / total as f32,
+                        }));
+                    };
+                    Ok(())
+                },
+            ) {
+                Err(e) => {
+                    error!("{e:?}");
+                    thread::sleep(RETRY_CONNECT_MIN_INTERVAL);
+                    continue;
```

Comment:
There are a few of these:
```
error!("{e:?}");
thread::sleep(RETRY_CONNECT_MIN_INTERVAL);
continue;
```
or slight variations. See if you can take most of the code you added in this function, and move it into the alvr_adb crate as a separate function that returns a Result. Then when handling the result you can use the 3 lines above a single time.